pkg/api/v1/types.go

```diff
@@ -982,9 +982,9 @@ type ServiceSpec struct {
 	// CreateExternalLoadBalancer indicates whether a load balancer should be created for this service.
 	CreateExternalLoadBalancer bool `json:"createExternalLoadBalancer,omitempty" description:"set up a cloud-provider-specific load balancer on an external IP"`
 
-	// PublicIPs are used by external load balancers, or can be set by
+	// Deprecated. PublicIPs are used by external load balancers, or can be set by
```

Comment:
I am wondering why are we marking it as deprecated and not just deleting it?
Since v1 has not been released yet, we can make breaking API changes.

Are we keeping the field to be able to do a round trip from v1beta1,2,3 -> internal -> v1 -> internal -> v1beta1,2,3?